src/ChunkData.cpp

```diff
@@ -46,103 +39,90 @@ cChunkData::cChunkData(cAllocationPool<cChunkData::sChunkSection> & a_Pool) :
 
 cChunkData::~cChunkData()
 {
-	#if __cplusplus < 201103L
-		// auto_ptr style interface for memory management
-		if (!m_IsOwner)
-		{
-			return;
-		}
-	#endif
-	for (size_t i = 0; i < NumSections; i++)
-	{
-		Free(m_Sections[i]);
-		m_Sections[i] = nullptr;
-	}
+	Clear();
 }
 
 
 
 
 
-#if __cplusplus < 201103L
-	// auto_ptr style interface for memory management
-	cChunkData::cChunkData(const cChunkData & a_Other) :
-		m_IsOwner(true),
-		m_Pool(a_Other.m_Pool)
+cChunkData::cChunkData(const cChunkData & a_Other):
```

Comment:
That doesn't work I'm afraid.  First copy creates a copy with the same pool, then if it gets moved into an object with a different pool it does _another_ copy.  In fact maybe the copy function should be removed altogether to avoid that happening.